packages/nimble-components/src/table/components/row/index.ts

```diff
@@ -20,13 +21,32 @@ export class TableRow<
     @observable
     public data?: TData;
 
-    // TODO: Temporarily assume the set of columns will be an array of strings.
-    // Eventually, this will be an array of column definitions.
     @observable
-    public columns: string[] = [];
+    public columns: TableColumn[] = [];
 
-    public getCellValue(column: string): TableFieldValue {
-        return this.data ? this.data[column] : undefined;
+    public getCellState(column: TableColumn): TableCellState<TableRecord> {
+        const fieldNames = column.getRecordFieldNames();
+        if (this.hasValidFieldNames(fieldNames) && this.data) {
+            const cellDataValues = fieldNames.map(field => this.data![field]);
```

Comment:
I would expect the non-null assertion to no longer be necessary (and in fact, causing a lint error).
```suggestion
            const cellDataValues = fieldNames.map(field => this.data[field]);
```